mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list.go

```diff
@@ -0,0 +1,415 @@
+	Timeouts         timeouts.Value `tfsdk:"timeouts"`
+}
+
+type ProjectIPAccessListRS struct {
+	client *MongoDBClient
+}
+
+func NewProjectIPAccessListRS() resource.Resource {
+	return &ProjectIPAccessListRS{}
+}
+
+var _ resource.Resource = &ProjectIPAccessListRS{}
+var _ resource.ResourceWithImportState = &ProjectIPAccessListRS{}
+
+func (r *ProjectIPAccessListRS) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, projectIPAccessList)
+}
+
+func (r *ProjectIPAccessListRS) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	client, err := ConfigureClient(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+
+	r.client = client
+}
+
+func (r *ProjectIPAccessListRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"cidr_block": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					cstmvalidator.ValidCIDR(),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("aws_security_group"),
+						path.MatchRelative().AtParent().AtName("ip_address"),
+					}...),
+				},
+			},
+			"ip_address": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					cstmvalidator.ValidIP(),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("aws_security_group"),
+						path.MatchRelative().AtParent().AtName("cidr_block"),
+					}...),
+				},
+			},
+			"aws_security_group": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("ip_address"),
+						path.MatchRelative().AtParent().AtName("cidr_block"),
+					}...),
+				},
+			},
+			"comment": schema.StringAttribute{
+				Computed: true,
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+			}),
```

Comment:
I see that is previous implementation we had resource timeout defined for Read and Delete, any particular reason for this change? Would removing the Delete attribute be a breaking change?

Reply:
the only reason is that I though it was create 🤦  Changing it to delete

Reply:
Updated